.github/workflows/test.yml

```diff
@@ -1,323 +1,85 @@
-                    "tests/end_to_end/http_server/sqlite",
-                    "tests/end_to_end/http_server/sqlite_drizzle",
-                    "tests/end_to_end/http_server/sqlite_typeorm",
-                    "tests/end_to_end/http_server/tfjs",
-                    "tests/end_to_end/http_server/web_assembly",
-                    "tests/property/ic_api/caller",
-                    "tests/property/ic_api/chunk",
-                    "tests/property/ic_api/id",
-                    "tests/property/ic_api/instruction_counter",
-                    "tests/property/ic_api/is_controller",
-                    "tests/property/ic_api/performance_counter",
-                    "tests/property/ic_api/time",
-                    "tests/property/ic_api/trap"
-                  ]
-                  END
-                  )
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//'%'/'%25'}"
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//$'\n'/'%0A'}"
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//$'\r'/'%0D'}"
-                  echo "::set-output name=example_directories::$EXAMPLE_DIRECTORIES"
-            - id: should_run_tests
+                  directories: |
+                      ./examples
+                      ./tests
+                  exclude_dirs: |
+                      ${{ env.IS_FEATURE == 'true' && format('
+                         ./examples/basic_bitcoin
+                         ./examples/bitcoin_psbt
+                         ./examples/ckbtc
+                         ./tests/end_to_end/http_server/ethers_base
+                         ./tests/end_to_end/http_server/http_outcall_fetch
+                         ./tests/end_to_end/http_server/ic_evm_rpc
+                         ./tests/property/candid_rpc/class_api/stable_b_tree_map
+                         ./tests/property/candid_rpc/functional_api/stable_b_tree_map
+                         ') || '' }}
+
+    determine-should-run-tests:
+        name: Determine if tests should run
+        runs-on: ubuntu-latest
+        outputs:
+            should_run_tests: ${{ steps.determine_should_run_tests.outputs.should_run_tests }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: determine_should_run_tests
+              # Determine if tests should run for this pr. If it's a manually submitted release candidate no tests will run.
               run: |
                   BRANCH_NAME="${{ github.head_ref }}"
                   RELEASE_VERSION="${BRANCH_NAME:9}"
                   COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")
-                  if [[ "${{ contains(github.head_ref, 'release--') }}" == "true" && "$COMMIT_MESSAGE" != "azle-bot automated release $RELEASE_VERSION" ]]
-                  then
-                    ./publish-github-action.sh $RELEASE_VERSION ${{ toJSON(steps.example_directories.outputs.example_directories) }}
+
+                  IS_AUTOMATED_RELEASE="false"
```

Comment:
This seems repeated...is there no way we can package this up somehow and only run it once?

Reply:
moved it into it's own action. I don't think there is any reason to prevent it from running twice, it's not that long of a task, and I'm not even sure it's possible for workflows to communicate with each other. But as an action the code isn't duplicated at least